cmd/kube-controller-manager/app/policy.go

```diff
@@ -67,6 +67,7 @@ func startDisruptionController(ctx ControllerContext) (http.Handler, bool, error
 		client,
 		ctx.RESTMapper,
 		scaleClient,
+		client.Discovery(),
```

Comment:
We've had problems in the past with discovery error handling in controllers causing the KCM/CCM to crash loop. Can you look at #73337 and add the disruption controller to the list of controllers we check for problems with discovery on init?